src/App.tsx

```diff
@@ -3,40 +3,93 @@ import { BrowserRouter, Route, Switch } from 'react-router-dom';
 import { ThemeProvider } from '@material-ui/core/styles';
 import theme from './theme';
 
-import { MainContentWrapper } from './components/MainContentWrapper/MainContentWrapper';
+// API
+import { client } from './services/clients/AwsAppSync';
+import { ApolloProvider, useQuery } from '@apollo/client';
+import {
+  getCurrentFeed,
+  FeedData,
+  FeedVariables,
+} from './services/queries/getCurrentFeed';
+
+// custom components
 import { Header } from './components/Header/Header';
+import { MainContentWrapper } from './components/MainContentWrapper/MainContentWrapper';
+import { HandleApiResponse } from './components/HandleApiResponse/HandleApiResponse';
 
+// pages
 import { AddStoryPage } from './pages/AddStoryPage';
 import { HomePage } from './pages/HomePage';
 import { NewTabLivePage } from './pages/newtab/NewTabLivePage';
 import { ProspectsListPage } from './pages/prospects/ProspectsListPage';
 
 function App(): JSX.Element {
-  // hardcoding feed for now
-  const feed = 'en-US';
+  /**
+   * I am assuming the authenticated user arrives "knowing" which
+   * feed is the default one for them (currentFeed: string).
+   *
+   * It would be brilliant if they already arrived with a Feed object (id, name),
+   * but for now I'm hardcoding the 'en-US' feed as the default one unless
+   * another feed has been provided in the URL.
+   */
+
+  // extract feed name if user goes straight to a subpage, i.e. /de-DE/prospects/
+  let feedName = window.location.pathname.split('/')[1];
+
+  // provide a temporary fallback option
+  feedName =
+    feedName.length && ['en-US', 'en-GB', 'de-DE'].includes(feedName)
+      ? feedName
+      : 'en-US';
+
+  // get feed object from the API
+  const { loading, error, data } = useQuery<FeedData, FeedVariables>(
```

Comment:
could it be this call that's resulting in network requests being made even in testing?

Reply:
That is the call I have been trying to mock! The misbehaving tests are in `App.test.tsx`. No solution so far. I have trawled through the Apollo code to make sure there are no network calls (there appear to be none: the MockedProvider mocks its own client and everything seems fine... Except it doesn't work for us). I have rewritten the tests to use `react-test-renderer` library so that my tests match the Apollo examples: https://www.apollographql.com/docs/react/development-testing/testing/ but this hasn't made any difference to the way they fail. There should be a reason, and probably a very simple one. I'll keep digging.